### DISCUSSION TITLE
Image updates without giving write access to flux

### DISCUSSION THREAD

**Discussion by ghost** [fluxcd/flux2#3012] [Q&A] [answered]
Because I don't really want to give flux write access to my git repository, I believe that I must manually update manifests when I would like to deploy a new image. I'm not sure however whether I am overcomplicating things.

Now, if I have the following repositories:

`~/work/flux-infra`

and

`~/work/application`

where inside of the application directory I have a `.github/workflows` workflow, that builds the container image and pushes it to the registry whenever I commit to a certain branch,

I would need to create a script that manually updates the tag version of the built image in that workflow file, and in the manifest in the flux-infra repository so that they remain in sync.

Can someone check that I've understood how this works and that this is what must be done if not using the automatic image update feature?

I also wonder whether I can have the GitHub Action just tag the commit (bump the latest tag), and use `image: ghcr.io/blah/application:latest` in the manifest. Does flux scan the container registry and reconcile the cluster if it finds a more recent tag to an image?

**Comment by stefanprodan** [comment, marked as answer]
Flux does not work with mutable tags such as `latest` because nothing changes in the Deployment spec, so Kubernetes will not rollout the new version. You can use a GitHub Action that bumps the version in Git using `sed` or any other tools that can replace strings in YAMLs.
